### PR TITLE
ci: add tripleo scenario testing

### PR DESCRIPTION
**Pull request by guits**
This should help to see earlier any failure in a tripleo deployment scenario.

Signed-off-by: Guillaume Abrioux <gabrioux@redhat.com>

### CODE REVIEW

File: tests/functional/centos/7/ooo-collocation/hosts [hunk 1]
```diff
@@ -0,0 +1,97 @@
+all:
+  vars:
+    admin_secret: AQASm39aAAAAABAAKlMButBjorDoNbciw3NMBw==
+    ceph_conf_overrides:
+      global: {mon_max_pg_per_osd: 3072, osd_pool_default_pg_num: 8, osd_pool_default_pgp_num: 1,
+        osd_pool_default_size: 1, rgw_keystone_accepted_roles: 'Member, admin', rgw_keystone_admin_domain: default,
+        rgw_keystone_admin_password: v3tcc6XvNhNF6zNDUHBvuu4YJ, rgw_keystone_admin_project: service,
+        rgw_keystone_admin_user: swift, rgw_keystone_api_version: 3, rgw_keystone_implicit_tenants: 'true',
+        rgw_keystone_url: 'http://192.168.2.11:5000', rgw_s3_auth_use_keystone: 'true'}
```
Comment (guits) [inline tests/functional/centos/7/ooo-collocation/hosts]: check about that ip

File: tests/functional/centos/7/ooo-collocation/hosts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    ceph_stable: true
+    cephfs: cephfs
+    cephfs_data: manila_data
+    cephfs_metadata: manila_metadata
+    cephfs_pools:
+    - {name: manila_data, pgs: 128}
+    - {name: manila_metadata, pgs: 128}
+    cluster_network: 192.168.96.0/24
+    containerized_deployment: true
+    devices: [/dev/loop3, /dev/loop4, /dev/loop5, /dev/loop6]
```
Comment (guits) [inline tests/functional/centos/7/ooo-collocation/hosts]: fix the devices here

File: tests/functional/centos/7/ooo-collocation/hosts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        rbd_children, allow rwx pool=volumes, allow rwx pool=backups, allow rwx pool=vms,
+        allow rwx pool=images, allow rwx pool=metrics'}
+    - {key: AQASm39aAAAAABAA0DhN14/TTIy8OxGaJL25RQ==, mds_cap: allow *, mgr_cap: allow
+        *, mode: '0600', mon_cap: 'allow r, allow command \\\"auth del\\\", allow
+        command \\\"auth caps\\\", allow command \\\"auth get\\\", allow command \\\"auth
+        get-or-create\\\"', name: client.manila, osd_cap: allow rw}
+    - {key: AQASm39aAAAAABAASkRRPrzTgqOllVQKBOWrPQ==, mgr_cap: allow *, mode: '0600',
+      mon_cap: allow rw, name: client.radosgw, osd_cap: allow rwx}
+    openstack_pools:
+    - {name: images, pg_num: 32, rule_name: ''}
```
Comment (guits) [inline tests/functional/centos/7/ooo-collocation/hosts]: fix pg_num here and below

File: tests/functional/centos/7/ooo-collocation/hosts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        rgw_keystone_admin_password: v3tcc6XvNhNF6zNDUHBvuu4YJ, rgw_keystone_admin_project: service,
+        rgw_keystone_admin_user: swift, rgw_keystone_api_version: 3, rgw_keystone_implicit_tenants: 'true',
+        rgw_keystone_url: 'http://192.168.2.11:5000', rgw_s3_auth_use_keystone: 'true'}
+    ceph_docker_image: ceph/daemon
+    ceph_docker_image_tag: tag-stable-3.0-luminous-centos-7
+    ceph_docker_registry: docker.io
+    ceph_mgr_docker_extra_env: -e MGR_DASHBOARD=0
+    ceph_origin: repository
+    ceph_repository: community
+    ceph_osd_docker_cpu_limit: 1
```
Comment (gfidente) [inline tests/functional/centos/7/ooo-collocation/hosts]: not sure if we want these two or are better left untouched?

File: tests/functional/centos/7/ooo-collocation/hosts [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    ceph_osd_docker_memory_limit: 3g
```
Comment (gfidente) [inline tests/functional/centos/7/ooo-collocation/hosts]: as above

File: tests/functional/centos/7/ooo-collocation/hosts [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+        get-or-create\\\"', name: client.manila, osd_cap: allow rw}
+    - {key: AQASm39aAAAAABAASkRRPrzTgqOllVQKBOWrPQ==, mgr_cap: allow *, mode: '0600',
+      mon_cap: allow rw, name: client.radosgw, osd_cap: allow rwx}
+    openstack_pools:
+    - {name: images, pg_num: 8, rule_name: ''}
+    - {name: metrics, pg_num: 8, rule_name: ''}
+    - {name: backups, pg_num: 8, rule_name: ''}
+    - {name: vms, pg_num: 8, rule_name: ''}
+    - {name: volumes, pg_num: 8, rule_name: ''}
+    osd_max_backfills: 1
```
Comment (gfidente) [inline tests/functional/centos/7/ooo-collocation/hosts]: as above

File: tests/functional/centos/7/ooo-collocation/hosts [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    osd_objectstore: filestore
+    osd_recovery_max_active: 3
```
Comment (gfidente) [inline tests/functional/centos/7/ooo-collocation/hosts]: as above

File: tests/functional/centos/7/ooo-collocation/hosts [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+    osd_recovery_op_priority: 3
```
Comment (gfidente) [inline tests/functional/centos/7/ooo-collocation/hosts]: as above

File: tests/functional/centos/7/ooo-collocation/hosts [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+    mon0: {}
+nfss:
+  hosts: {}
+osds:
+  hosts:
+    mon0: {}
+rbdmirrors:
+  hosts: {}
+rgws:
+  hosts: {}
```
Comment (gfidente) [inline tests/functional/centos/7/ooo-collocation/hosts]: do I get it right this won't be deploying RGW?

Comment (guits) [reply tests/functional/centos/7/ooo-collocation/hosts]: you are right, this won't deploy a rgw.
this is what I got from @fultonj should we add a rgw ? collocated with mon0 in that case ?

File: tests/functional/centos/7/ooo-collocation/hosts [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+  hosts:
+    mon0: {}
+mons:
+  hosts:
+    mon0: {}
+nfss:
+  hosts: {}
+osds:
+  hosts:
+    mon0: {}
```
Comment (andrewschoen) [inline tests/functional/centos/7/ooo-collocation/hosts]: I'm unsure if the CI will allow us to collocate an OSD with a MON. The MON nodes do not get the same device setup and private/public network as the OSD nodes do.

File: tests/functional/centos/7/ooo-collocation/hosts (hunk 9 above)
Comment (gfidente) [reply tests/functional/centos/7/ooo-collocation/hosts]: here is a full list of params (which includes rgw): http://ix.io/N4q

File: tests/functional/centos/7/ooo-collocation/hosts (hunk 10 above)
Comment (fultonj) [reply tests/functional/centos/7/ooo-collocation/hosts]: That's OK, we don't need the OSD collocated with the mon.

**Comment by guits** [comment]
@andrewschoen I agree.
related PR: ceph/ceph-build#960

**Comment by guits** [comment]
jenkins test all

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-bluestore_osds_container

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-ooo_collocation

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-bluestore_osds_non_container

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-purge_bluestore_osds_container

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-purge_filestore_osds_container

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-filestore_osds_container

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-update_docker_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-switch_to_containers

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-shrink_osd_container

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-shrink_osd_container

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-shrink_osd

**Comment by guits** [comment]
jenkins test dev-ansible2.4-bluestore_lvm_osds

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-ooo_collocation

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-ooo_collocation

**Pull request merged**